### DISCUSSION TITLE
Unparenthesized `SubSelect` as `FromItem`

### DISCUSSION THREAD

**Discussion by ssteinhauser** [JSQLParser/JSqlParser#2071] [Ideas]
As already stated in my first contribution to this project, I'd like to implement some features for [Exasol ](https://www.exasol.com/) DBMS.
The first thing was to support unparenthesized sub selects (see #2046).

However, one related feature is still open for discussion: support of unparenthesized sub select as `FromItem`

> I've had a look at `LateralSubSelect`, but I think to support the complete syntax of the feature flagged unparenthesised `Select()` as `FromItem`, the `Select` class would have to implement the `FromItem` interface and I would have to implement the methods of `FromItem` in all sub classes which are extending `Select`.
> Before I go ahead and do the implementation I would like to ask for your opinion on this. Do you agree with it or do you have another suggestion.

_Originally posted by @ssteinhauser in https://github.com/JSQLParser/JSqlParser/issues/2046#issuecomment-2248022362_

I'd like to discuss this topic here to work out a solution I could implement then.

**Comment by manticore-projects** [comment]
Greetings and welcome back.

Please just go ahead, I would be here in case of any questions. Cheers!

**Reply by ssteinhauser** [reply]
Alright, thank you. I've just provided a PR (#2073) - please let me know your thoughts.

**Comment by ssteinhauser** [comment]
Discussion is closed with PR #2073
